pkg/kubelet/dockershim/docker_sandbox.go

```diff
@@ -324,65 +324,75 @@ func (ds *dockerService) RemovePodSandbox(ctx context.Context, r *runtimeapi.Rem
 	return nil, utilerrors.NewAggregate(errs)
 }
 
-// getIPFromPlugin interrogates the network plugin for an IP.
-func (ds *dockerService) getIPFromPlugin(sandbox *dockertypes.ContainerJSON) (string, error) {
+// getIPsFromPlugin interrogates the network plugin for an IP.
+func (ds *dockerService) getIPsFromPlugin(sandbox *dockertypes.ContainerJSON) ([]string, error) {
 	metadata, err := parseSandboxName(sandbox.Name)
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 	msg := fmt.Sprintf("Couldn't find network status for %s/%s through plugin", metadata.Namespace, metadata.Name)
 	cID := kubecontainer.BuildContainerID(runtimeName, sandbox.ID)
 	networkStatus, err := ds.network.GetPodNetworkStatus(metadata.Namespace, metadata.Name, cID)
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 	if networkStatus == nil {
-		return "", fmt.Errorf("%v: invalid network status for", msg)
+		return nil, fmt.Errorf("%v: invalid network status for", msg)
+	}
+
+	ips := make([]string, 0)
+	for _, ip := range networkStatus.IPs {
+		ips = append(ips, ip.String())
+	}
+	// if we don't have any ip in our list then cni is using classic primary IP only
+	if len(ips) == 0 {
+		ips = append(ips, networkStatus.IP.String())
```

Comment:
No.PodStatus is not without `IP` not on internal type and not on v1.

Reply:
I am sorry, I am having trouble parsing that remark, can you please elaborate on what you mean?

I am confused by what I find in commit f19a8df101b998be2c63d8e73c63e266e5a0f7c4 (the latest on your `ipv6dualstack` branch).  In https://github.com/khenidak/kubernetes/blob/f19a8df101b998be2c63d8e73c63e266e5a0f7c4/staging/src/k8s.io/api/core/v1/types.go I see that the external type for PodStatus has both `PodIP` and `PodIPs`, while in https://github.com/khenidak/kubernetes/blob/f19a8df101b998be2c63d8e73c63e266e5a0f7c4/pkg/apis/core/types.go I see that the internal type has only `PodIPs`.  I recognize that the internal and external types do not have to be the same, but I suspect your remark is saying they _are_ the same.

It is difficult to evaluate this piece of code because its boundary conditions have incorrect comments.  The comment on this function is outdated, it says that it returns one IP address.  On the other end, this code is consuming a k8s.io/kubernetes/pkg/kubelet/dockershim/network.PodNetworkStatus, which has both `IP` and `IPs`, and the comment on the latter suggests it does not contain the value in `IP`, and none of the comments say anything about divergent content when "cni is using classic primary IP only".